src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2474,7 +2484,13 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           if (result0.nonEmpty) checkPure(result0, supple = true)
         }
 
-        treeCopy.Block(block, statsTyped, expr1)
+        // Remove ValDef for right-associative by-value operator desugaring which has been inlined into expr1
+        val statsTyped2 = statsTyped match {
+          case (vd: ValDef) :: Nil if inlinedRightAssocValDefs contains vd.symbol => Nil
+          case _ => statsTyped
+        }
+
+        treeCopy.Block(block, statsTyped2, expr1)
```

Comment:
This leaves a single-expression block in place, but I guess that's fine. We'd have to handle it outside `typedBlock`, as this method returns a `Block`.

```
scala> def foo = { 1 f_: c; 2 }
[[syntax trees at end of                     typer]] // <console>

        def foo: Int = {
          {
            $line5.$read.$iw.$iw.c.f_:(1)
          };
          2
        }
```